### PR TITLE
Rename Slip10 to AccountKeyDeriver

### PR DESCRIPTION
**Pull request by voloshyn**
### Motivation

It's not immediately apparent what `Slip10` class is used for, renaming it to `AccountKeyDeriver` will clearly state what's the purpose of this class.

relevant android-bindings changes: https://github.com/mobilecoinfoundation/fog/pull/42

### In this PR
* Rename `Slip-10` to `AccountKeyDeriver`
* add `all_english_words_are_correct` mnemonic unit test

### Future Work
* Default Android code formatter is pretty bad, especially when formatting long strings, need to find a better solution.

### CODE REVIEW

File: android-sdk/src/main/java/com/mobilecoin/lib/AccountKeyDeriver.java [hunk 1]
```diff
@@ -6,8 +6,8 @@
 
 import com.mobilecoin.lib.exceptions.BadBip39EntropyException;
 
-class Slip10 extends Native {
-    private final static String TAG = Slip10.class.getName();
+class AccountKeyDeriver extends Native {
```
Comment (samdealy) [inline android-sdk/src/main/java/com/mobilecoin/lib/AccountKeyDeriver.java:9]: Can you add a class JavaDoc?

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/MnemonicTest.java [hunk 2]
```diff
@@ -88,9 +92,28 @@ public void words_by_prefix() throws BadMnemonicException {
         assertArrayEquals(new String[]{"zebra", "zero", "zone", "zoo"}, words);
     }
 
+    @Test
+    public void all_english_words_are_correct()
+            throws NoSuchAlgorithmException, BadMnemonicException {
+        assertEquals(
+                "ffbc2f3228ee610ad011ff9d38a1fb8e49e23fb60601aa7605733abb0005b01e",
```
Comment (kylefleming) [inline android-sdk/src/androidTest/java/com/mobilecoin/lib/MnemonicTest.java:99]: Good idea. Is this value part of the BIP-39 spec somewhere?

**Review by kylefleming — APPROVED** [review]

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/MnemonicTest.java [hunk 3]
```diff
@@ -44,41 +51,38 @@ public void create_mnemonic_12_words() throws BadEntropyException {
     public void create_mnemonic_24_words() throws BadEntropyException {
         String mnemonic = Mnemonics.bip39EntropyToMnemonic(Hex.toByteArray(
                 "6a36844131c70de644643935ac6610445d04adca6670f6bb6ee46935c873e214"));
-        assertEquals("health reflect aware glory ignore veteran " +
-                "bag mango cup glimpse lottery master " +
-                "space finger civil sock wall swarm " +
-                "ribbon sponsor frame delay marriage oyster", mnemonic);
+        assertEquals("health reflect aware glory ignore veteran bag mango cup glimpse lottery " +
+                "master space finger civil sock wall swarm ribbon sponsor frame delay marriage " +
+                "oyster", mnemonic);
```
Comment (samdealy) [inline android-sdk/src/androidTest/java/com/mobilecoin/lib/MnemonicTest.java:56]: Consider moving the expected value string into a local variable. Might make it more readable / easier to update in the future.

**Review by samdealy — APPROVED** [review]

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/MnemonicTest.java (hunk 2 above)
Comment (voloshyn) [reply android-sdk/src/androidTest/java/com/mobilecoin/lib/MnemonicTest.java:99]: No, it was provided by Alan

Comment (voloshyn) [reply android-sdk/src/androidTest/java/com/mobilecoin/lib/MnemonicTest.java:99]: Or maybe it is, but I didn't see it in the test vectors

File: android-sdk/src/main/java/com/mobilecoin/lib/AccountKeyDeriver.java (hunk 1 above)
Comment (voloshyn) [reply android-sdk/src/main/java/com/mobilecoin/lib/AccountKeyDeriver.java:12]: done

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/MnemonicTest.java (hunk 3 above)
Comment (voloshyn) [reply android-sdk/src/androidTest/java/com/mobilecoin/lib/MnemonicTest.java:56]: Let's find a solution to formatting strings and then refactor this. For now, I'll keep it like this.

**Pull request merged**